cdap-app-fabric/src/main/java/co/cask/cdap/app/stream/StreamWriterFactory.java

```diff
@@ -20,14 +20,17 @@
 import co.cask.cdap.proto.Id;
 import com.google.inject.assistedinject.Assisted;
 
+import java.util.List;
+
 /**
  * Factory to create {@link StreamWriter} objects
  */
 public interface StreamWriterFactory {
   /**
-   * @param programId the programId that will be using the {@link StreamWriter}
+   * @param namespace the namespace that the {@link StreamWriter} belongs to
+   * @param owners the owners of the {@link StreamWriter}
    * @return a {@link StreamWriter} for the specified namespaceId
    */
-  StreamWriter create(@Assisted("programId") Id.Program programId);
+  StreamWriter create(@Assisted("namespace") Id.Namespace namespace, @Assisted("owners") List<Id> owners);
```

Comment:
I can use `NamespacedId`, but seems odd when I have to pick one `Id.Namespace` from the list? Maybe I can just have some check to verify that they're all in the same namespace.